### PR TITLE
Remove reference import from support and remove the feature flag

### PR DESCRIPTION
**Pull request by C-gyorfi**
## Context

We are replacing the google form solution for references with a DFE form, therefore we no longer need the `import reference` feature on the `support console`, neither the `reference_form` feature flag.

## Changes proposed in this pull request
This PR removes:
-  The `import reference` feature from the support interface and all related dead code 💀 
- `Reference form` feature flag

## Guidance to review

Trust

## Link to Trello card

https://trello.com/c/ycutK8Mx/634-remove-reference-import-from-support-and-remove-the-feature-flag

## Things to check
- [x] This code doesn't rely on migrations in the same Pull Request
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/mailers/referee_mailer.rb [hunk 1]
```diff
@@ -4,15 +4,10 @@ def reference_request_email(application_form, reference)
...
-      @refuse_to_give_feedback_link = referee_interface_refuse_feedback_url(token: unhashed_token)
-      template_name = :reference_request_email
-    else
-      @reference_link = google_form_url_for(@candidate_name, @reference)
-      template_name = :reference_request_by_google_form_email
-    end
+    unhashed_token = reference.update_token!
+    @reference_link = referee_interface_reference_feedback_url(token: unhashed_token)
+    @refuse_to_give_feedback_link = referee_interface_refuse_feedback_url(token: unhashed_token)
+    template_name = :reference_request_email
```
Comment (alexdesi) [inline app/mailers/referee_mailer.rb]: I suppose we do not need anymore the `template_name` variable

File: spec/mailers/referee_mailer_spec.rb [hunk 2]
```diff
@@ -201,4 +119,3 @@
     end
   end
 end
-# rubocop:enable RSpec/NestedGroups
```
Comment (tvararu) [inline spec/mailers/referee_mailer_spec.rb]: ❤️

File: app/mailers/referee_mailer.rb [hunk 3]
```diff
@@ -26,11 +21,8 @@ def reference_request_chaser_email(application_form, reference)
     @candidate_name = application_form.full_name
 
     @token = reference.update_token!
-    @reference_link = if FeatureFlag.active?('reference_form')
-                        referee_interface_reference_feedback_url(token: @token)
-                      else
-                        google_form_url_for(@candidate_name, @reference)
-                      end
+    @reference_link = referee_interface_reference_feedback_url(token: @token)
```
Comment (alexdesi) [inline app/mailers/referee_mailer.rb]: Maybe you can move `referee_interface_reference_feedback_url(token: @token)` in the view

File: spec/mailers/referee_mailer_spec.rb [hunk 4]
```diff
@@ -27,7 +25,6 @@
 
     describe 'with reference_form feature flag on' do
```
Comment (tijmenb) [inline spec/mailers/referee_mailer_spec.rb]: The `describe` here can be removed (it's always true)

File: spec/mailers/referee_mailer_spec.rb [hunk 5]
```diff
@@ -36,46 +33,6 @@
...
-      end
-
-      it 'sends an email with a link to a prefilled Google form' do
-        body = mail.body.encoded
-        expect(body).to include(t('reference_request.google_form_url'))
-        expect(body).to include("=#{reference.id}")
-        expect(body).to include("=#{CGI.escape(reference.email_address)}")
-      end
-
-      it 'encodes spaces as %20 rather than + in the Google form parameters for correct prefilling' do
```
Comment (tijmenb) [inline spec/mailers/referee_mailer_spec.rb]: Good riddance to this test!

File: spec/mailers/referee_mailer_spec.rb (hunk 4 above)
Comment (C-gyorfi) [reply spec/mailers/referee_mailer_spec.rb]: Good catch!

Comment (C-gyorfi) [reply spec/mailers/referee_mailer_spec.rb]: Resolved in c7abaa5

File: app/mailers/referee_mailer.rb (hunk 3 above)
Comment (C-gyorfi) [reply app/mailers/referee_mailer.rb]: Resolved in c7abaa5

**Pull request merged**